### DISCUSSION TITLE
How to use the next router in layout.tsx? For conditional rendering based on authentication

### DISCUSSION THREAD

**Discussion by jasonrdunne** [vercel/next.js#59421] [App Router]
When any page attempts to be rendered, i first want to make sure that the user is signed in. This is my root layout; for simplicity's sake i abstracted rerouting logic authwrapper:

```
export default function RootLayout({ children }: Props) {
  return (
    <html lang='en'>
      <body>
        <Configure />
        <Navigation />
        <AuthProvider>
          <AuthWrapper>
            {children}
          </AuthWrapper>
        </AuthProvider>
      </body>
    </html>
  );
}
```

In auth wrapper:

```
const AuthWrapper: React.FC<AuthWrapperProps> = ({ children }) => {
  const { authState } = useAuth();
  const router = useRouter();

  useEffect(() => {
    if (!authState.isLoading && !authState.accessToken && router.pathname !== '/') {
      router.push('/');
    }
  }, [authState, router]);

  if (authState.isLoading) {
    return <div>Loading...</div>;
  }

  return <>{children}</>;
};

export default AuthWrapper;
```

As you can see, I'm just rerouting if the authentication failed. However, it gives me this error:

```
Error: NextRouter was not mounted. https://nextjs.org/docs/messages/next-router-not-mounted

  10 | const AuthWrapper: React.FC<AuthWrapperProps> = ({ children }) => {
  11 | const { authState } = useAuth();
> 12 | const router = useRouter();
```

I would much rather keep this check in layout.tsx because i want it to apply to every page by default. I don't want to have to add the same routing logic to every page.

**Comment by bklaing2** [comment]
[`useRouter`](https://nextjs.org/docs/app/building-your-application/routing/linking-and-navigating#userouter-hook) can only be used in client components. You need to add the `'use client'` directive at the top of the `AuthWrapper` file and make sure you're importing `useRouter` from `next/navigation`
